### PR TITLE
Make sure that XHR uses the XDomainRequest on old browsers.

### PR DESCRIPTION
**Pull request by gkatsev**

### CODE REVIEW

**Comment by heff** [comment]
How will this affect same-domain tracks?

**Comment by gkatsev** [comment]
Hm... Good question. XHR uses regular XmlHttpRequests in modern browsers, so, for those, we won't be affected. Looking into how XDR works to figure out whether it should be an issue as well as trying some out.

**Comment by gkatsev** [comment]
I'm going to open an issue against XHR as well.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: c8a8ed5290ea3f5ac882ef85dc37fe044fd22922
Build details: https://travis-ci.org/pam/video.js/builds/81824626

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by gkatsev** [comment]
Also, sounds like XDR might work with same domain as long as the CORS headers are set up correctly.

**Comment by gkatsev** [comment]
It works, still need to confirm whether it'll require CORS headers or not.

**Comment by gkatsev** [comment]
Yep, XDR requires the CORS headers even for same-domain requests.

**Comment by heff** [comment]
If this doesn't introduce same-domain issues, then looks good to me.

**Comment by gkatsev** [comment]
It does mean we can't request tracks from the same domain if it doesn't also include CORS.

**Comment by gkatsev** [comment]
Btw, here's the issue against XHR: https://github.com/Raynos/xhr/issues/94

**Comment by gkatsev** [comment]
I'm going to add https://github.com/videojs/video.js/blob/stable/src/js/xhr.js#L80-L84 in here for now so it won't be a blocker and then we can discuss raynos/xhr#94 there and whether a PR for this will be submitted.

**Comment by gkatsev** [comment]
Looks like that doesn't specific snippet doesn't take into account protocol relative urls. But I know how to handle those too.

**Comment by gkatsev** [comment]
@heff a more complete imlementation now.

**Comment by pam** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: b5718b979fa5632b3c5019e75dd7e0f725c572f8
Build details: https://travis-ci.org/pam/video.js/builds/81849412

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by pam** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 25f26ace61d03d498c3efbd919d6829e7681651c
Build details: https://travis-ci.org/pam/video.js/builds/81850165

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: src/js/tracks/text-track.js [hunk 1]
```diff
@@ -253,7 +254,23 @@ var parseCues = function(srcContent, track) {
...
 
 var loadTrack = function(src, track) {
-  XHR({ uri: src }, Fn.bind(this, function(err, response, responseBody){
+  let urlInfo = parseUrl(src);
+  let winLoc = window.location;
+  // IE8 protocol relative urls will return ':' for protocol
+  let srcProtocol = urlInfo.protocol === ':' ? winLoc.protocol : urlInfo.protocol;
+  // Check if url is for another domain/origin
+  // IE8 doesn't know location.origin, so we won't rely on it here
+  let crossOrigin = (srcProtocol + urlInfo.host) !== (winLoc.protocol + winLoc.host);
```
Comment (gkatsev) [inline src/js/tracks/text-track.js]: maybe I should move that into our `url` module.

**Comment by heff** [comment]
lgtm

**Comment by heff** [comment]
lgtm!

File: test/unit/utils/url.test.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import document from 'global/document';
 import window from 'global/window';
 import * as Url from '../../../src/js/utils/url.js';
+import proxyquireify from 'proxyquireify';
+const proxyquire = proxyquireify(require);
```
Comment (gkatsev) [inline test/unit/utils/url.test.js]: Proxyquireify uses a module called `merge-descriptors` that is currently throwing some warnings in chrome. I'll open an issue against it so we can get that closed out.

Comment (gkatsev) [inline test/unit/utils/url.test.js]: Opened an issue: https://github.com/component/merge-descriptors/issues/5

**Comment by gkatsev** [comment]
Somehow completely missed the jshint issue.

File: src/js/utils/url.js [hunk 3]
```diff
@@ -95,3 +96,25 @@ export const getFileExtension = function(path) {
...
+ * @method isCrossOrigin
+ */
+export const isCrossOrigin = function(url) {
+  let urlInfo = parseUrl(url);
+  let winLoc = window.location;
+
+  // IE8 protocol relative urls will return ':' for protocol
+  let srcProtocol = urlInfo.protocol === ':' ? winLoc.protocol : urlInfo.protocol;
+
+  console.log(srcProtocol, winLoc.protocol, urlInfo.host, winLoc.host);
```
Comment (gkatsev) [inline src/js/utils/url.js]: this is for travis testing :/. Will be removed ASAP.

**Pull request closed without merging**